frontend-react/src/components/USLink.tsx

```diff
@@ -17,25 +16,51 @@ type USLinkProps = AnchorHTMLAttributes<{}> &
     Omit<CustomLinkProps, "activeClassName">;
 type USNavLinkProps = Pick<AnchorHTMLAttributes<{}>, "href"> & CustomLinkProps;
 
+/**
+ * Stateless function to get route href from href that could be
+ * absolute, shorthand, relative, and/or a non-route.
+ * Attempt to parse href as URL (taking into account "//" shorthand).
+ * If it errors, then assume its a relative url (aka route). If it
+ * parses, then verify its an absolute route through origins.
+ */
+export function getHrefRoute(href?: string): string | undefined {
+    if (href === undefined) return undefined;
```

Comment:
I think this check might be redundant since it'll be caught as an error and returned [below](https://github.com/CDCgov/prime-reportstream/pull/8254/files#diff-ca9c2c77fb047cac23c0b9a0b5740bc1f961ac5fcc83bea960f6537c3ff341ccR39), no?

Reply:
TypeScript correctly gets angry at possibly passing undefined to URL, which is not allowed.

Reply:
Oh, damn you perfectly correct TypeScript!  Ignore me then :)